shared-settings.gradle

```
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import org.yaml.snakeyaml.Yaml
+
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+
+    dependencies {
+        classpath 'org.yaml:snakeyaml:1.33'
+    }
+}
+
+// Synchronized library configuration for all modules
+// This "componentsVersion" number is defined in "version.txt" and should follow
+// semantic versioning (MAJOR.MINOR.PATCH). See https://semver.org/
+class Config {
+
+    public final String componentsVersion
+    public final String componentsGroupId
+    public final Integer compileSdkVersion
+    public final Integer minSdkVersion
+    public final Integer targetSdkVersion
+
+    Config(
+            String componentsVersion,
+            String componentsGroupId,
+            Integer compileSdkVersion,
+            Integer minSdkVersion,
+            Integer targetSdkVersion
+    ) {
+        this.componentsVersion = componentsVersion
+        this.componentsGroupId = componentsGroupId
+        this.compileSdkVersion = compileSdkVersion
+        this.minSdkVersion = minSdkVersion
+        this.targetSdkVersion = targetSdkVersion
+    }
+}
+
+def setupProject(name, path, description) {
+    settings.include(":$name")
+
+    project(":$name").projectDir = new File(rootDir, "../android-components/${path}")
+
+    // project(...) gives us a skeleton project that we can't set ext.* on
+    gradle.beforeProject { project ->
+        // However, the "afterProject" listener iterates over every project and gives us the actual project
+        // So, once we filter for the project we care about, we can set whatever we want
+        if (project.name == name) {
+            project.ext.description = description
+        }
+    }
+}
+
+def yaml = new Yaml()
+def buildconfig = yaml.load(new File(rootDir, '../android-components/.buildconfig.yml').newInputStream())
+
+buildconfig.projects.each { project ->
+    // If we're building A-C let's setup all project, otherwise exclude samples e.g., if we're building Fenix or Focus.
```

```suggestion
    // If we're building A-C, setup all projects including samples. Otherwise, exclude samples e.g., if we're building Fenix or Focus.
```
I think I needed this extra clarity to fully spell it out for me.

---

Ya, I know we added this change, but for whatever reason when I was just comparing it all, I needed it the extra wording for it to fully click.